internal/pkg/cli/app_init.go

```diff
@@ -58,97 +64,116 @@ type InitAppOpts struct {
 
 	// Outputs stored on successful actions.
 	manifestPath string
-
-	*GlobalOpts
 }
 
-// Ask prompts for fields that are required but not passed in.
-func (opts *InitAppOpts) Ask() error {
-	if opts.AppType == "" {
-		if err := opts.askAppType(); err != nil {
-			return err
-		}
+func newInitAppCmd(opts *initAppOpts) (*initAppCmd, error) {
```

Comment:
Overall LGTM. Do we need a test func for this? Since decoupling initialization means testing becomes possible.

Reply:
Changed it up :)